app/v1/services/RetrieveCharitableGivingReliefService.scala

```diff
@@ -60,4 +60,16 @@ class RetrieveCharitableGivingReliefService @Inject() (connector: RetrieveCharit
       "SERVICE_UNAVAILABLE"     -> InternalError
     )
 
+    val extraTysErrors = Map(
+      "INVALID_TAX_YEAR"             -> TaxYearFormatError,
+      "INVALID_CORRELATION_ID"       -> InternalError,
+      "INVALID_INCOMESOURCE_ID"      -> InternalError,
+      "INVALID_INCOMESOURCE_TYPE"    -> InternalError,
+      "SUBMISSION_PERIOD_NOT_FOUND"  -> NotFoundError,
+      "INCOME_DATA_SOURCE_NOT_FOUND" -> NotFoundError,
+      "TAX_YEAR_NOT_SUPPORTED"       -> RuleTaxYearNotSupportedError
+    )
+    errors ++ extraTysErrors
```

Comment:
Kind of needs a blank line above this, as it's a bit bunched into the extraTysErrors block